### ISSUE TITLE
Type error with callback argument to generic method

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#1273]
**Describe the bug**

Pyright is unable to handle the type of a generic method that expects a callback function that uses the generic type.

**Screenshots or Code**
```python
from typing import Callable, Generic, Optional, Tuple, TypeVar

Msg = TypeVar("Msg")
Reply = TypeVar("Reply")

class AsyncReplyChannel(Generic[Reply]):
    ...

class MailboxProcessor(Generic[Msg]):
    def post_and_async_reply(self, build_message: Callable[[AsyncReplyChannel[Reply]], Msg]) -> Optional[Reply]:
        return None

agent: MailboxProcessor[Tuple[int, AsyncReplyChannel[str]]] = MailboxProcessor()
build_message: Callable[[AsyncReplyChannel[str]], Tuple[int, AsyncReplyChannel[str]]] = lambda r: (42, r)
ret = agent.post_and_async_reply(build_message)

reveal_type(ret)
```

I think that this should be correctly typed since the generic type `Reply` is present both in input and output of the generic method, but I might be expecting too much, since the argument is a callable. Pyright gives the error:

<img width="1011" alt="Screenshot 2020-12-12 at 13 57 13" src="https://user-images.githubusercontent.com/849479/101984527-3eaef300-3c82-11eb-8720-8c5dd58b2715.png">

**VS Code extension or command-line**

vscode extension: 1.1.94

**Additional context**

mypy does not complain and seems to get the return type correctly:

<img width="667" alt="Screenshot 2020-12-12 at 13 57 46" src="https://user-images.githubusercontent.com/849479/101984524-39ea3f00-3c82-11eb-8bc8-939672f2e833.png">

This screenshot was taken when running pylance language server 2020.12.1 (pyright 4ef6f2b2) in addition to mypy (0.790).

**Comment by erictraut** [comment]
Thanks, this is a bug indeed. It will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.95, which I just published. It will also be included in the next release of pylance.

**Comment by dbrattli** [comment]
Confirmed that this release fixes the issue. Thanks for fixing it!
